pages/marriage.php

```diff
@@ -149,9 +149,9 @@ function marriage() {
     if(isset($_POST['confirm_divorce'])) {
         try {
             $system->query("UPDATE `users` SET `spouse`='0', `marriage_time`='0' WHERE `user_id`='$player->spouse' LIMIT 1");
```

Comment:
Added new commit for this. This will work only change I made is to intval the spouse info so the type comparison works. The only thing I will note from my testing is this won't clear any 3 way mixed up marriages. A -> B, B -> C, C -> A. In those cases it will only clear out the person who clicked divorce. The other two people will still be married on their side but will need to do the divorce as well. I left the other code commented out in case we want to bring it back in. That will check all cases and clear everything that is invalid.

Reply:
That makes sense. I'd be open to that but we have to make sure cases like the orphaned marriage are fixable (see other comment)

Reply:
Makes sense. I can add some more comments into that other code to see if I can get it to make sense or if you have any other input for it. The code that I had is designed to check the player to determine if the marriage is correct in multiple cases.

1. Check the spouse of the player to determine who they are married to. If married to the player it's a correct marriage and divorce is handled as intended. 
2. If the spouse is married to someone else, we check the spouse's spouse (this would be the second chain of marriage) if that matches we ignore that marriage.
3. If that marriage is also invalid, we clear that chain of marriages. (player's spouse and the user married to them).
4. We do a final check to see if anyone is married to the player, if so we clear that as well.

This will clear any 3 link marriages (A -> B, B -> C, C -> A) it won't handle anything more messed up than that but I did pretty extensive testing before in my dev environment to ensure it won't clear valid marriages between players.